pysurveycto/pysurveycto.py

```diff
@@ -724,3 +727,70 @@ def list_forms(self):
             raise e
         
         return response.json()['forms']
+
+    def upload_dataset(self, df, dataset_id, dataset_title=None, append=False, fill=False):
+        """
+        Uploads a pandas df to a dataset
+        :return: dictionary with previous dataset preview and upload response
+
+        :param df: pandas DataFrame to upload
+        :param dataset_id: ID of the dataset on SurveyCTO
+        :param dataset_title: Optional title for the dataset (defaults to dataset_id)
+        :param append: If True, appends data; otherwise replaces the dataset
+        :param fill: If True, allows mismatched columns in append mode
+        
+        """
+        
+        assert isinstance(df, pd.DataFrame), "data must be a pandas DataFrame"
+        assert isinstance(dataset_id, str), "dataset_id must be a string"
+        if dataset_title is None:
+            dataset_title = dataset_id
```

Comment:
If the dataset_title is not needed for the API, I think we can remove this from the params as well. There is no check that the title is correct and seems redundant. What do you you think?